csi/exporter.py

```diff
@@ -1,129 +1,142 @@
-            'kadalu_storage_pv_capacity_used_bytes',
-            'Kadalu Storage PV Used Capacity',
-            labels=capacity_labels+["pv"]
-        )
-        pv_capacity_free_bytes = GaugeMetricFamily(
-            'kadalu_storage_pv_capacity_free_bytes',
-            'Kadalu Storage PV Free Capacity',
-            labels=capacity_labels+["pv"]
-        )
-
-        for dirname in os.listdir(HOSTVOL_MOUNTDIR):
-            labels = [dirname]  # TODO: Add more labels
-            pth = os.path.join(HOSTVOL_MOUNTDIR, dirname)
-            if os.path.ismount(pth):
-                stat = os.statvfs(pth)
-
-                # Capacity
-                total = stat.f_bsize * stat.f_blocks
-                free = stat.f_bsize * stat.f_bavail
-                used = total - free
-                capacity_bytes.add_metric(labels, total)
-                capacity_free_bytes.add_metric(labels, free)
-                capacity_used_bytes.add_metric(labels, used)
-
-                # Inodes
-                total = stat.f_files
-                free = stat.f_favail
-                used = total - free
-                inodes_count.add_metric(labels, total)
-                inodes_free_count.add_metric(labels, free)
-                inodes_used_count.add_metric(labels, used)
-
-                # Gathers capacity metrics for each subvol
-                for pvc in yield_pvc_from_mntdir(os.path.join(pth, "info")):
-                    # pvc[0]: name
-                    # pvc[1]: size
-                    # pvc[2]: path prefix
-                    pvcpath = os.path.join(pth, pvc[2])
-                    pvcname = pvc[0]
-                    pvclabels = labels + [pvcname]
-
-                    stat = os.statvfs(pvcpath)
-
-                    # Capacity
-                    total = stat.f_bsize * stat.f_blocks
-                    free = stat.f_bsize * stat.f_bavail
-                    used = total - free
-                    pv_capacity_bytes.add_metric(pvclabels, total)
-                    pv_capacity_free_bytes.add_metric(pvclabels, free)
-                    pv_capacity_used_bytes.add_metric(pvclabels, used)
-
-        yield capacity_bytes
-        yield capacity_free_bytes
-        yield capacity_used_bytes
-        yield inodes_count
-        yield inodes_free_count
-        yield inodes_used_count
-        yield pv_capacity_bytes
-        yield pv_capacity_used_bytes
-        yield pv_capacity_free_bytes
-
-
-REGISTRY.register(CsiMetricsCollector())
+app = FastAPI()
+
+@app.get("/_api/metrics")
+def metrics():
+    """
+    Gathers storage and pvcs metrics.
+    Starts process by exposing the data collected in port 8050 at '/_api/metrics'.
+    """
+
+    data = {
+        "pod": {},
+        "storages": []
+    }
+
+    pod_name = ""
+    memory_usage_in_bytes = 0
+    cpu_usage_in_nanoseconds = 0
+
+    pod_name_path = '/etc/hostname'
+    with open(pod_name_path, 'r') as pod_fd:
+        pod_name = pod_fd.read().strip()
```

Comment:
Operator can set this name after calling API of each Pod. You can remove this from here.